src/crimson/os/seastore/lba_manager.h

```diff
@@ -252,6 +252,224 @@ class LBAManager {
+    const LBAIter &iter) = 0;
+
+  /**
+   * next_iterator
+   *
+   * The wrapper of underlying fixed kv btree, return the next iterator of
+   * provided iterator.
+   */
+  virtual make_iterator_ret next_iterator(
+    Transaction &t,
+    const LBAIter &iter) = 0;
+
+  /**
+   * lower_bound
+   *
+   * The wrapper of underlying FixedKVBtree, return the least iterator whose
+   * key >= laddr
+   */
+  virtual make_iterator_ret lower_bound(
+    Transaction &t,
+    laddr_t laddr) = 0;
+
+  /**
+   * upper_bound
+   *
+   * The wrapper of underlying FixedKVBtree, return the least iterator whose
+   * key > laddr
+   */
+  virtual make_iterator_ret upper_bound(
+    Transaction &t,
+    laddr_t laddr) = 0;
+
+  /**
+   * upper_bound_right
+   *
+   * The wrapper of underlying FixedKVBtree, return the least iterator whose
+   * key + val.length > laddr
+   */
+  virtual make_iterator_ret upper_bound_right(
+    Transaction &t,
+    laddr_t laddr) = 0;
+
+  /**
+   * find_region
+   *
+   * Search for a free region which can hold the specified length
+   * of lba mappings.
+   * Returns the starting laddr of the found region and an iterator
+   * that can be used to insert mappings.
+   */
+  struct find_region_result_t {
+    find_region_result_t()
+	: laddr(L_ADDR_NULL), iter() {}
+    find_region_result_t(laddr_t laddr, LBAIter &&iter)
+	: laddr(laddr), iter(iter) {}
+    laddr_t laddr;
+    LBAIter iter;
+  };
+  using find_region_iertr = base_iertr;
+  using find_region_ret = find_region_iertr::future<
+    find_region_result_t>;
+  virtual find_region_ret find_region(
+    Transaction &t,
+    laddr_t hint,
+    extent_len_t length) = 0;
+
+  /**
+   * get_iterator
+   *
+   * Retrieves the iterator for the laddr. If the lba mapping is indirect,
+   * the returned iterator will not resolve the direct mapping.
+   */
+  using get_iterator_ret = get_mapping_iertr::future<LBAIter>;
+  virtual get_iterator_ret get_iterator(
+    Transaction &t,
+    laddr_t laddr) = 0;
+
+  /**
+   * get_iterators
+   *
+   * retrieves the iterators in range [laddr, laddr + length). The indirect
+   * mappings are not resolved.
+   */
+  using get_iterators_ret = get_mappings_iertr::future<std::vector<LBAIter>>;
+  virtual get_iterators_ret get_iterators(
+    Transaction &t,
+    laddr_t laddr,
+    extent_len_t length) = 0;
+
+  /**
+   * insert_mapping
+   *
+   * Insert a new pair of key value before the specified LBAIter.
+   * Return the iterator points to the inserted mapping.
+   */
+  using insert_mapping_iertr = base_iertr;
+  using insert_mapping_ret = insert_mapping_iertr::future<LBAIter>;
+  virtual insert_mapping_ret insert_mapping(
+    Transaction &t,
+    const LBAIter &iter,
+    laddr_t laddr,
+    lba_map_val_t value,
+    LogicalCachedExtent *nextent) = 0;
+
+  /**
+   * change_mapping
+   *
+   * Change the mapping at the provided LBAIter to the new key value pair.
+   * nextent should be reserved ptr if the original extent is retired.
```

Comment:
"nextent should be reserved ptr if the original extent is retired"

I'm not fully following this part yet, is it related to `RemappedExtentPlaceholder` ?

Reply:
`change_mapping` has some use cases:
1. `update_mapping`, update the paddr of lba mapping in normal IO path;
2. remap lba mapping
    a. change the length of lba mapping
    b. change the lba maping to its sub-mapping
3. make lba mapping indirect. the original lba mapping is direct and change the pladdr to laddr. It is used by "pin head" feature.

Seems we should divide it into
1. `update_paddr`, accepting not null `nextent`;
2. `remap_mapping`, accepting not null `nextent`;
3. `make_indirect`, not requiring `nextent`  parameter;

Reply:
Thanks! Sounds better than the original one.